tests/unit/models/governing-body-test.js

```diff
@@ -0,0 +1,828 @@
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2024-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should not be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no error when the model's start date is the same as the end date of another governing body", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2024-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2024-01-01'),
+        endDate: new Date('2025-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should not be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no errors when the model's start date is strictly within the period of another governing body and they have the same end date", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2024-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-06-06'),
+        endDate: new Date('2024-01-01'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should not be undefined');
+
+      otherBodiesStub.restore();
+    });
+
+    test("it returns no errors when the model's end date is strictly within the period of another governing body and they have the same start date", async function (assert) {
+      const otherModel = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2024-01-01'),
+      });
+
+      const model = this.store().createRecord('governing-body', {
+        startDate: new Date('2023-01-01'),
+        endDate: new Date('2023-06-06'),
+      });
+
+      const otherBodiesStub = sinon.stub(model, 'getOtherTimedGoverningBodies');
+      otherBodiesStub.resolves([otherModel]);
+
+      const isValid = await model.validate();
+
+      assert.true(isValid);
+      assert.notOk(model.error, 'model.error should not be undefined');
+
+      otherBodiesStub.restore();
+    });
```

Comment:
**thought:** All these tests looks really close, isn't possible to group them in a for loop?

Reply:
I merged these together in single test with loop.